src/rdiff_backup/connection.py

```diff
@@ -363,6 +366,23 @@ def _close(self):
         """Close the pipes associated with the connection"""
         self.outpipe.close()
         self.inpipe.close()
+        # reap the pipe child with wait() to ensure any final output from
+        # the pipe by commands that run after rdiff-backup server; otherwise
+        # a race condition occurs where final output is sometimes lost;
+        # Python>=3.3 gives the timeout option to wait: set to a modestly
+        # small value here to minimize possibility of introducing bugs/delays
+        if self.process:
+            try:
+                self.process.wait(timeout=5)
+            except subprocess.TimeoutExpired:
+                # if some longer delay occurred, just kill it, try hard so
+                # we avoid the ResourceWarning if it's still running when
+                # the subprocess destructors hit; yes ugly sleeps, but rare
+                self.process.terminate()
```

Comment:
Add please a warning log message that "the server process {sp} is being terminated".format(sp=self.process.args)